packages/shared/src/hooks/post/useBlockPostPanel.ts

```diff
@@ -0,0 +1,273 @@
+  { toastOnSuccess, blockedSource }: UseBlockPostProps = {},
+): UseBlockPost => {
+  const { openModal } = useLazyModal();
+  const { displayToast } = useToastNotification();
+  const { onBlockTags, onUnfollowSource, onUnblockTags, onFollowSource } =
+    useTagAndSource({
+      origin: Origin.TagsFilter,
+      postId: post?.id,
+      shouldInvalidateQueries: false,
+    });
+  const client = useQueryClient();
+  const { user } = useAuthContext();
+  const { checkHasCompleted, completeAction } = useActions();
+  const key = generateStorageKey(
+    StorageTopic.Post,
+    `block:${post?.id}`,
+    user?.id,
+  );
+  const { data } = useQuery<BlockData>(key, () => client.getQueryData(key), {
+    initialData: {},
+    ...disabledRefetch,
+  });
+  const setShowTagsPanel = useCallback(
+    (params: BlockData) =>
+      client.setQueryData<BlockData>(key, (current) => ({
+        ...current,
+        ...params,
+      })),
+    [client, key],
+  );
+  const { reportPost } = useReportPost();
+  const { trackEvent } = useAnalyticsContext();
+
+  const value = useMemo<BlockData>(
+    () => ({
+      ...data,
+      showTagsPanel: checkHasCompleted(ActionType.HideBlockPanel)
+        ? undefined
+        : data?.showTagsPanel,
+    }),
+    [data, checkHasCompleted],
+  );
+
+  const updateFeedPreferences = useCallback(
+    async (
+      blocks: string[],
+      unblocks: string[],
+      shouldBlockSource?: boolean,
+    ) => {
+      const onUpdateSource = shouldBlockSource
+        ? onUnfollowSource
+        : onFollowSource;
+
+      const results = await Promise.all([
+        blocks.length ? onBlockTags({ tags: blocks }) : ignoredCall(),
+        unblocks.length ? onUnblockTags({ tags: unblocks }) : ignoredCall(),
+        isNullOrUndefined(shouldBlockSource)
+          ? ignoredCall()
+          : onUpdateSource({ source: post.source }),
+      ]);
+
+      return results.every(({ successful }) => successful);
+    },
+    [onFollowSource, onUnfollowSource, onBlockTags, onUnblockTags, post],
+  );
+
+  const onUndo = useCallback(
+    async (blocks: string[], shouldBlockSource?: boolean) => {
+      const successful = await updateFeedPreferences(
+        [],
+        blocks,
+        shouldBlockSource,
+      );
+
+      if (!successful) return;
+
+      setShowTagsPanel({
+        showTagsPanel: undefined,
+        blocked: {},
+      });
+    },
+    [setShowTagsPanel, updateFeedPreferences],
+  );
+
+  const onClose = useCallback(
+    (forceClose?: boolean) =>
+      setShowTagsPanel({ showTagsPanel: forceClose ? undefined : false }),
+    [setShowTagsPanel],
+  );
+
+  const onShowPanel = useCallback(
+    () => setShowTagsPanel({ showTagsPanel: true }),
+    [setShowTagsPanel],
+  );
+
+  const onDismissPermanently = useCallback(() => {
+    completeAction(ActionType.HideBlockPanel);
+    setShowTagsPanel({ showTagsPanel: undefined });
+  }, [completeAction, setShowTagsPanel]);
+
+  const onReport = useCallback(() => {
```

Comment:
Just a generic comment, we now use a mix of lazy modal and the direct modal, I'd suggest maybe making one generic usage if possible.
So converting the old one to use lazyModal as well.

And if that's the case we can probably move this onReport logic to a custom hook so it can be handled from one place? 

I'd prefer this in case we decide to add something (tags is good example) or tracking needs an addition change.

Love to hear what you think.

Reply:
Agreed. After checking, it seems we can actually make it simpler by executing the report post from the modal instead of having a callback. We do keep the callback though, but only for cases where you need it, and not should be a required prop from the modal itself.

Reply:
Looks great, great modification 👍